lib/getNewThresholds.js

```diff
@@ -9,7 +9,7 @@ const getNewThresholds = (thresholds, coverages, margin, tolerance) =>
     if (desiredCoverage >= threshold + margin) {
```

Comment:
I think we should calculate the rounded/truncated `desiredCoverage` value _before_ this condition, otherwise we may end up updating the number to the same value.

Reply:
Im not quite following what you mean about calculating the number first. could you just explain in a bit more detail please

Reply:
We may need to calculate the final `desiredCoverage` value (accounting for the truncation) _before_ entering this `if` block.

Let's say `desiredCoverage == 50.25`, `threshold == 50.2`, `margin == 0`. Assuming we support custom precision, and it's `1` in this case, we actually don't need to update coverage thresholds, because the new value would be the same as the current one: `50.2`.

So the math you do below must be done earlier, likely together with `tolerance`:

```js
const desiredCoverage = Math.trunc((coverage - tolerance) * factor) / factor
```

In fact, we should probably replace the `>=` with `>` as well to avoid a `0` bump, but that can be left to a separate PR.